src/main/java/com/limechain/transaction/TransactionState.java

```diff
@@ -0,0 +1,119 @@
+package com.limechain.transaction;
+
+import com.limechain.transaction.dto.Extrinsic;
+import com.limechain.transaction.dto.ValidTransaction;
+import com.limechain.utils.ByteArrayUtils;
+import lombok.extern.java.Log;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+@Log
+@Component
+public class TransactionState {
+
+    private final TransactionPool transactionPool;
+    private final ExecutorService executor;
+    private final Queue<ValidTransaction> transactionQueue;
+
+    public TransactionState() {
+        transactionPool = new TransactionPool();
+        executor = Executors.newSingleThreadExecutor();
+        transactionQueue = new PriorityQueue<>();
+    }
+
+    public void pushTransaction(ValidTransaction validTransaction) {
+        transactionQueue.add(validTransaction);
+    }
+
+    public ValidTransaction popTransaction() {
```

Comment:
Not a big deal, but since 'pop' semantic suggests a stack-related operation, it might be better to rename it to pollTransaction for clarity.